### ISSUE TITLE
Block with nesting cannot be change easily with Modifier.setBlockType

### ISSUE THREAD

**Issue by SamyPesse** [mitermayer/draft-js#15]
In the current Draft examples, The toolbar uses [Modifier.setBlockType](https://facebook.github.io/draft-js/docs/api-reference-modifier.html#setblocktype) to change a block type.

But if list item should have nested `unstyled` block, changing a paragraph into a list item will not create the nested `unstyled` block.

We can fix in [splitNestedBlockInContentState](https://github.com/SamyPesse/draft-js/blob/feature/tree/src/model/transaction/splitNestedBlockInContentState.js), so that:

``` js
{
   type: 'a-block-type-withnesting'
   text: 'Current text'
}

-> Transform to ->

{
    type: 'a-block-type-withnesting'
    text: '',
    blocks: [
          {
               type: 'unstyled',
               text: 'Current text'
          },
          {
               type: 'unstyled',
               text: '<cursor>'
          }
    ]
}
```

Basically, we should consider that a block has some text or some nested blocks, but it should not have both.

**Comment by mitermayer** [comment]
agreed

**Comment by mitermayer** [comment]
Will start working on this one, should not be too hard

**Comment by mitermayer** [comment]
Started this in here https://github.com/SamyPesse/draft-js/commit/7a527e4a1eb0c257d80b42f7583c6e79b33faf99 

Still need to make sure that when we press ENTER in one of those blocks we treat as if it was root element.

example: 

li -> h1

press enter

should become

li -> h1
li ->

**Comment by mitermayer** [comment]
Thats is done and working now
